From 1d7b7046c8f735e150e92aeace3fe6d0686b9bc9 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Wed, 3 Oct 2018 10:52:29 +0200
Subject: [PATCH] Fix version.h inclusion to be from kernel build folder

Instead of relying on Toolchain headers which can be from newer kernel.

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 Makefile_G1G2 | 3 +++
 Makefile_H1   | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/Makefile_G1G2 b/Makefile_G1G2
index 29866a4..c473bcb 100755
--- a/Makefile_G1G2
+++ b/Makefile_G1G2
@@ -16,6 +16,9 @@ INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/include/uapi -I$(LINUX_KERNEL_ROOT)/in
 # ION header location
 INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/drivers/staging/android/uapi
 
+# LINUX_VERSION_CODE from kernel build folder instead of toolchain headers
+INCLUDE_HEADERS += -I$(LINUX_KERNEL_BUILD)/include/generated/uapi
+
 CFLAGS += -DDEC_MODULE_PATH=\"/dev/mxc_hantro\" -DUSE_FAKE_RFC_TABLE -DFIFO_DATATYPE=void* -DNDEBUG -DDOWN_SCALER \
            -DUSE_EXTERNAL_BUFFER -DUSE_FAST_EC -DUSE_VP9_EC -DGET_FREE_BUFFER_NON_BLOCK \
            -DDEC_X170_OUTPUT_FORMAT=0 -DDEC_X170_TIMEOUT_LENGTH=-1 -DENABLE_HEVC_SUPPORT \
diff --git a/Makefile_H1 b/Makefile_H1
index 56b4332..0be43ce 100755
--- a/Makefile_H1
+++ b/Makefile_H1
@@ -23,6 +23,9 @@ ENV += -I$(SDKTARGETSYSROOT)/usr/imx/include
 # ION header location
 ENV += -I$(LINUX_KERNEL_ROOT)/drivers/staging/android/uapi
 
+# LINUX_VERSION_CODE from kernel build folder instead of toolchain headers
+ENV += -I$(LINUX_KERNEL_BUILD)/include/generated/uapi
+
 LIBENCNAME = libcodec_enc
 LIBSENC = -L./ -lhantro_h1 -lpthread
 
-- 
2.19.0

